### ISSUE TITLE
fileserver.clear_lock not clearing pillar locks (v2015.8.1)

### ISSUE THREAD

**Issue by bradthurber** [saltstack/salt#28298]
I noticed this error in master debug log
```
[WARNING ] Update lockfile is present for git_pillar remote 'master git@bitbucket.org:bthurber/salt-master-pillar-db.git', skipping. If this warning persists, it is possible that the update process was interrupted. Removing /var/cache/salt/master/git_pillar/ff758f3207d2061300390d06413309ef/update.lk or running 'salt-run fileserver.clear_lock git_pillar' will allow updates to continue for this remote.
```

I run the command as directed and it (incorrectly) doesn't remove any locks
```
[root@s-cdc-jbsalt-01 salt]# salt-run fileserver.clear_lock git_pillar
No locks were removed
```

Removing the file manually makes the error go away
```
rm /var/cache/salt/master/git_pillar/ff758f3207d2061300390d06413309ef/update.lk
```

I now see this in the master log instead
```
[DEBUG   ] Set lock for git@bitbucket.org:bthurber/salt-master-pillar-db.git
[DEBUG   ] git_pillar is fetching from 'master git@bitbucket.org:bthurber/salt-master-pillar-db.git'
[DEBUG   ] git_pillar remote 'master git@bitbucket.org:bthurber/salt-master-pillar-db.git' is up-to-date
[DEBUG   ] pygit2 does not support detecting stale refs for authenticated remotes, saltenvs will not reflect branches/tags removed from remote 'master git@bitbucket.org:bthurber/salt-master-pillar-db.git'
[DEBUG   ] Removed lock for git@bitbucket.org:bthurber/salt-master-pillar-db.git
```

...and the pillar files are in the cache
```
[root@s-cdc-jbsalt-01 datasources]# pwd
/var/cache/salt/master/git_pillar/ff758f3207d2061300390d06413309ef/datasources
[root@s-cdc-jbsalt-01 datasources]# ls -la
total 16
drwxr-xr-x 2 root root 4096 Oct 22 16:45 .
drwxr-xr-x 4 root root 4096 Oct 26 09:53 ..
-rw-r--r-- 1 root root  274 Oct 22 16:45 dev-int1.sls
-rw-r--r-- 1 root root  274 Oct 22 16:45 dev-int2.sls
```

Unfortunately, there still seems to be a problem somewhere as gitfs pillars do not work. I have a feeling that the failure to clear the lock and the gitfs pillars not working are probably related so here's some additional info...: 
```
[root@s-cdc-jbsalt-01 pillar]# salt 'd*app*' pillar.item _errors
d-cdc-jbapp-21.adesa.com:
    ----------
    _errors:
        - Specified SLS 'datasources.dev-int2' in environment 'base' is not available on the salt master
```

pillar top.sls excerpt
```
base:
...
  'G@environment:dev-int2 and G@roles:jboss':
    - match: compound
    - datasources.dev-int2
...
```

ext_pillar config
```
[root@s-cdc-jbsalt-01 master.d]# cat extpillar.conf
git_pillar_branch: master
git_pillar_provider: pygit2
git_pillar_privkey: /etc/salt/master.d/id_rsa
git_pillar_pubkey: /etc/salt/master.d/id_rsa.pub

ext_pillar:
  - git:
    - master git@bitbucket.org:bthurber/salt-master-pillar-db.git:
      - env: base

```

salt-master --versions
```
[root@s-cdc-jbsalt-01 salt]# salt-master --versions               
Salt Version:                                                     
           Salt: 2015.8.1                                         
                                                                  
Dependency Versions:                                              
         Jinja2: 2.8                                              
       M2Crypto: 0.20.2                                           
           Mako: 0.3.4                                            
         PyYAML: 3.11                                             
          PyZMQ: 14.5.0                                           
         Python: 2.6.6 (r266:84292, Nov 21 2013, 10:50:32)        
           RAET: Not Installed                                    
        Tornado: 4.2.1                                            
            ZMQ: 4.0.5                                            
           cffi: Not Installed                                    
       cherrypy: Not Installed                                    
       dateutil: 1.4.1                                            
          gitdb: Not Installed                                    
      gitpython: Not Installed                                    
          ioflo: Not Installed                                    
        libnacl: Not Installed                                    
   msgpack-pure: Not Installed                                    
 msgpack-python: 0.4.6                                            
   mysql-python: Not Installed                                    
      pycparser: Not Installed                                    
       pycrypto: 2.6.1                                            
         pygit2: 0.20.3                                           
   python-gnupg: Not Installed                                    
          smmap: Not Installed                                    
        timelib: Not Installed                                    
                                                                  
System Versions:                                                  
           dist: redhat 6.6 Santiago                              
        machine: x86_64                                           
        release: 2.6.32-504.el6.x86_64                            
         system: Red Hat Enterprise Linux Server 6.6 Santiago     
```

My _**minions**_ are running an older version of salt (either 2015.5.5 or 2015.5.3). This particular minion that I'm using as my test case is 2015.5.5. My master was running 2015.5.5 as well before I upgraded it to 2015.8.1 - as I wanted to try gitfs pillars with pygit2.

**Comment by bradthurber** [comment]
More info: 

If I run `salt 'd*app-21*' saltutil.refresh_pillar` I see this in the salt master log.

```
[ERROR   ] Specified SLS 'datasources.dev-int2' in environment 'base' is not available on the salt master
[DEBUG   ] compile template: False
[ERROR   ] Template was specified incorrectly: False
[DEBUG   ] LazyLoaded git.ext_pillar
[CRITICAL] Exception caught while initializing git_pillar remote 'master git@bitbucket.org:bthurber/salt-master-pillar-db.git': tuple index out of range
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/salt/utils/gitfs.py", line 261, in __init__
    self.new = self.init_remote()
  File "/usr/lib/python2.6/site-packages/salt/utils/gitfs.py", line 783, in init_remote
    log.error(_INVALID_REPO.format(self.cachedir, self.url))
IndexError: tuple index out of range
[ERROR   ] Failed to load ext_pillar git: Failed to load git_pillar
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/salt/pillar/__init__.py", line 631, in ext_pillar
    key)
  File "/usr/lib/python2.6/site-packages/salt/pillar/__init__.py", line 588, in _external_pillar_data
    pillar_dirs)
  File "/usr/lib/python2.6/site-packages/salt/pillar/git_pillar.py", line 245, in ext_pillar
    pillar.init_remotes(repo, PER_REMOTE_OVERRIDES)
  File "/usr/lib/python2.6/site-packages/salt/utils/gitfs.py", line 1566, in init_remotes
    self.role
  File "/usr/lib/python2.6/site-packages/salt/utils/gitfs.py", line 679, in __init__
    override_params, cache_root, role)
  File "/usr/lib/python2.6/site-packages/salt/utils/gitfs.py", line 268, in __init__
    failhard(self.role)
  File "/usr/lib/python2.6/site-packages/salt/utils/gitfs.py", line 107, in failhard
    raise FileserverConfigError('Failed to load {0}'.format(role))
FileserverConfigError: Failed to load git_pillar
[CRITICAL] Pillar render error: Specified SLS 'datasources.dev-int2' in environment 'base' is not available on the salt master
```

In addition, I'm now getting this error each time it polls the repo (every 60 seconds) for changes

```
[DEBUG   ] Set lock for git@bitbucket.org:bthurber/salt-master-pillar-db.git
[DEBUG   ] git_pillar is fetching from 'master git@bitbucket.org:bthurber/salt-master-pillar-db.git'
[ERROR   ] Exception 'Reference 'HEAD' not found' caught while fetching git_pillar remote 'master git@bitbucket.org:bthurber/salt-master-pillar-db.git'
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/salt/utils/gitfs.py", line 1702, in fetch_remotes
    if repo.fetch():
  File "/usr/lib/python2.6/site-packages/salt/utils/gitfs.py", line 882, in fetch
    fetch_results = origin.fetch()
GitError: Reference 'HEAD' not found
[DEBUG   ] Removed lock for git@bitbucket.org:bthurber/salt-master-pillar-db.git
[DEBUG   ] Could not LazyLoad config.merge
```

Full disclosure, my _**minions**_ are running an older version of salt (either 2015.5.5 or 2015.5.3). This particular minion that I'm using as my test case is 2015.5.5. My master was running 2015.5.5 as well before I upgraded it to 2015.8.1 - as I wanted to try gitfs pillars with pygit2.

**Comment by Ch3LL** [comment]
@bradthurber  This is a duplicate of the following issue #27738 which has been fixed with #27802 . Thanks for the report.

**Comment by bradthurber** [comment]
@Ch3LL Thank you for the update
